pkgs/tools/package-management/nix/default.nix

```diff
@@ -225,79 +225,71 @@ common =
     sha256 = "sha256-SPnam4xNIjbMgnq6IP1AaM1V62X0yZNo4DEVmI8sHOo=";
   };
 
-in rec {
+  buildNix =
```

Comment:
It's not clear to me what the advantage of adding *another* function is, given that we already have the `common` function. Can't we just move any commonality into that one, rather than adding another layer of indirection?